packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/synthesize-resources.ts

```diff
@@ -142,29 +140,36 @@ export const removeDeprecatedProps = (props: $TSObject): $TSObject => {
   return props;
 };
 
-const lambdaTriggers = async (coreAnswers: $TSObject, context: $TSContext, previouslySaved: $TSAny) => {
+const lambdaTriggers = async (coreAnswers: $TSObject, context: $TSContext, previouslySaved: $TSAny): Promise<void> => {
   const { handleTriggers } = await import('./trigger-flow-auth-helper');
   let triggerKeyValues = {};
   let authTriggerConnections: AuthTriggerConnection[];
   if (coreAnswers.triggers) {
     const triggerConfig = (await handleTriggers(context, coreAnswers, previouslySaved)) as AuthTriggerConfig;
     triggerKeyValues = triggerConfig.triggers;
     authTriggerConnections = triggerConfig.authTriggerConnections;
+
+    // eslint-disable-next-line no-param-reassign
```

Comment:
We have 176 of that in the repo. Is this rule useful for us ?

Reply:
Good point.. I think it is a decent rule; but it requires a lot of restructuring to implement. For example, you would need to return a new value from this function instead of modifying the param; which is good, for immutability, but I also think its a much larger refactor

Reply:
Yeah, that's what I though when I saw this function. We should probably make a case to require it for new code. But untangling existing pattern would be separate chunk of work.